pilz_testutils/src/async_test.cpp

```diff
@@ -43,10 +44,20 @@ void AsyncTest::barricade(std::initializer_list<std::string> clear_events)
                [this](std::string event){ return this->waitlist_.count(event) == 0; });
   waitlist_.clear();
 
+  auto end_time_point = std::chrono::system_clock::now() + std::chrono::milliseconds(timeout_ms);
+
   while(!clear_events_.empty())
   {
-    cv_.wait(lk);
+    if (timeout_ms < 0)
+    {
+      cv_.wait(lk);
+    }
+    else if (cv_.wait_for(lk, end_time_point - std::chrono::system_clock::now()) == std::cv_status::timeout)
+    {
+      return clear_events_.empty();
+    }
```

Comment:
I think it would be more readlable:
```suggestion
    else 
    {
      std::future_status status = cv_.wait_for(lk, end_time_point - std::chrono::system_clock::now())
      if (status == std::cv_status::timeout)
      {
        return clear_events_.empty();
      }
    }
```

Reply:
Thanks. I would have committed it, but you accidently used `future_status` instead of `cv_status` ...